usecase/usecase-embedding-android-tests/embeddingapi-async/src/org/xwalk/embedded/api/asyncsample/XWalkViewWithClearCacheForSingleFileAsync.java

```diff
@@ -0,0 +1,209 @@
+        .append("2. Click 'Clear Baidu Cache Single File' button.\n")
+        .append("Expected Result:\n\n")
+        .append("Test passes if the cache size decrease after step 2.");
+        new  AlertDialog.Builder(this)
+        .setTitle("Info" )
+        .setMessage(mess.toString())
+        .setPositiveButton("confirm" ,  null )
+        .show();
+        
+        mXWalkView.setResourceClient(new ResourceClient(mXWalkView));
+        mXWalkView.setUIClient(new UIClient(mXWalkView));
+        
+        cacheSizeTextView = (TextView) findViewById(R.id.text1);
+
+        clearBaiduButton = (Button) findViewById(R.id.clear_baidu_btn);
+        clearBaiduButton.setOnClickListener(new OnClickListener() {
+			
+			@Override
+			public void onClick(View v) {
+				// TODO Auto-generated method stub
+				if (mXWalkView != null) {
+					for (String image : imageUrls) {
+						mXWalkView.clearCacheForSingleFile(image);
+					}
+	    			cacheSizeTextView.setText("Cache Size: "+countCacheSize());
+				}
+			}
+		});
+        
+        loadBaiduButton = (Button) findViewById(R.id.baidu_btn);
+        loadBaiduButton.setOnClickListener(new OnClickListener() {
+			
+			@Override
+			public void onClick(View v) {
+				// TODO Auto-generated method stub
+				if (mXWalkView != null) {
+					mXWalkView.load("http://image.baidu.com/", null);
+				}				
+			}
+		});
+    }
+    
+    class UIClient extends XWalkUIClient{
+
+		public UIClient(XWalkView view) {
+			super(view);
+			// TODO Auto-generated constructor stub
+		}
+
+		@Override
+		public void onPageLoadStopped(XWalkView view, String url,
+				LoadStatus status) {
+			// TODO Auto-generated method stub
+			super.onPageLoadStopped(view, url, status);
+			cacheSizeTextView.setText("Cache Size: "+countCacheSize());
+		}		
+	}
+    
+    class ResourceClient extends XWalkResourceClient {
+
+        public ResourceClient(XWalkView xwalkView) {
+            super(xwalkView);
+        }
+
+        public WebResourceResponse shouldInterceptLoadRequest(XWalkView view, String url) {
+            Log.d(TAG, "Intercept load request: " + url);
+            imageUrls.add(url);
+            return super.shouldInterceptLoadRequest(view, url);
+        }
+    }    
+
+	private static String countCacheSize(){
+    	long blockSize = 0;
+    	try {
+        	File cacheDir = new File(EXTRA_CACHE_PATH);
+        	if (cacheDir != null) {
+    			if (cacheDir.isDirectory()) {
+    				blockSize = getFileSizes(cacheDir);
+    			} else {
+    				blockSize = getFileSize(cacheDir);
+    			}
+    		}
+		} catch (Exception e) {
+			// TODO: handle exception
+			Log.d(TAG, "count Cache Size error!");
+		}
+    	return FormetFileSize(blockSize);
+    }
+    
+    private static long getFileSize(File f) throws Exception{
+		// TODO Auto-generated method stub
+    	long size = 0;
+    	if (f.exists()) {
+    	     FileInputStream fis = new FileInputStream(f);
+    	     size = fis.available();
```

Comment:
http://stackoverflow.com/questions/116574/java-get-file-size-efficiently
stream.available() does not return the file length. It returns the amount of bytes which are available for read without blocking other streams. It is not necessarily the same amount of bytes as file length. To get the real length from a stream, you really need to read it (and count the read bytes meanwhile)

How about use the File or URL method to get the filesize